pkg/services/ngalert/schedule/schedule.go

```diff
@@ -506,7 +520,7 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 					currentFingerprint = f
 					if isPaused {
 						logger.Debug("Skip rule evaluation because it is paused")
-						return nil
+						return
```

Comment:
this changes semantics and basically cancels (unshedules) the ruleRoutine, so when rule is unpaused, nothing will happen.